noir_stdlib/src/array/mod.nr

```diff
@@ -43,10 +43,10 @@ impl<T, let N: u32> [T; N] {
     /// assert_eq(b, [2, 4, 6]);
     /// ```
     pub fn map<U, Env>(self, f: fn[Env](T) -> U) -> [U; N] {
-        let first_elem = f(self[0]);
+        let first_elem = crate::mem::zeroed();
         let mut ret = [first_elem; N];
```

Comment:
Maybe rename `first_elem` to something else, given that it's no longer the first element? Or maybe inline `crate::mem::zeroed()` in the array literal.